### DISCUSSION TITLE
How can I load custom font cross project?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#10609] [Q&A] [answered]
Currently I am migrating on a project from WPF to Avalonia.

There are projects A and B in the same solution. 

Here is the font: https://fonts.google.com/noto/specimen/Noto+Sans+SC

It is located on /Fonts/ in project A with 6 files.
NotoSansSC-Black.otf
NotoSansSC-Bold.otf
NotoSansSC-Light.otf
NotoSansSC-Medium.otf
NotoSansSC-Regular.otf
NotoSansSC-Thin.otf

I referenced project A in project B.

I am about to set it as a default font.

In WPF I set the font as Resource and declare it like this. Soon set it as default font in project B:
`<FontFamily x:Key="NotoSans">pack://A;component/Fonts/#Noto Sans</FontFamily>`

In Avalonia it seems I can only set it in the MainWindow of B:
`<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="B.MainWindow" FontFamily="avares://A/Fonts/#Noto Sans"
        Title="Test">	
</Window>`

After the program ran, VS reports an error:

> System.InvalidOperationException: 'Could not create glyph typeface for: Noto Sans.'

What's wrong with my code?

**Comment by Gillibald** [comment, marked as answer]
Your font files need to be added as AvaloniaResource

**Reply by kyuranger** [reply]
> Your font files need to be added as AvaloniaResource
It works now! By the way, there are so many dissections about font cannot be load in Linux, such as https://github.com/AvaloniaUI/Avalonia/issues/5136

Is this bug fixed now? Or I have to write a font manager as it said.

**Reply by timunie** [reply]
which Avalonia version are you using? Also please show us how you added the font to your cs-project. Is the build type AvaloniaResource?

**Reply by kyuranger** [reply]
> which Avalonia version are you using? Also please show us how you added the font to your cs-project. Is the build type AvaloniaResource?

I am using the newest:11.0.0-preview5. I added it by build type AvaloniaResources and reference it in fontfamily of the XAML MainWindow.

**Reply by timunie** [reply]
If the issue is still present for you, any chance you can share a sample App with us to double-check what may be the root cause?

**Reply by kyuranger** [reply]
> If the issue is still present for you, any chance you can share a sample App with us to double-check what may be the root cause?

Thank you. I found it is because there is some problem with my code, not a bug in Avalonia.
